barretenberg/cpp/src/barretenberg/vm2/tracegen/keccakf1600_trace.test.cpp

```diff
@@ -0,0 +1,372 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "barretenberg/vm2/common/aztec_constants.hpp"
+#include "barretenberg/vm2/common/constants.hpp"
+#include "barretenberg/vm2/common/memory_types.hpp"
+#include "barretenberg/vm2/simulation/events/keccakf1600_event.hpp"
+#include "barretenberg/vm2/simulation/keccakf1600.hpp"
+#include "barretenberg/vm2/testing/keccakf1600_fixture.test.hpp"
+#include "barretenberg/vm2/testing/macros.hpp"
+#include "barretenberg/vm2/tracegen/keccakf1600_trace.hpp"
+#include "barretenberg/vm2/tracegen/test_trace_container.hpp"
+
+namespace bb::avm2::tracegen {
+namespace {
+
+using ::testing::Field;
+
+using R = TestTraceContainer::Row;
+using C = Column;
+
+simulation::KeccakF1600Event create_standard_event_mem_slice()
+{
+    simulation::KeccakF1600Event event;
+    event.src_addr = 100;
+    event.dst_addr = 200;
+    event.space_id = 23;
+
+    // Initialize source memory values with U64 tags and state values
+    for (size_t i = 0; i < 5; i++) {
+        for (size_t j = 0; j < 5; j++) {
+            const uint64_t input_value = (i * 5 + j) + 1000;
+            const uint64_t output_value = (i * 5 + j) + 10000;
+            event.src_mem_values[i][j] = MemoryValue::from<uint64_t>(input_value);
+            event.rounds[0].state[i][j] = input_value;
+            if (i == 0 && j == 0) {
+                event.rounds[AVM_KECCAKF1600_NUM_ROUNDS - 1].state_iota_00 = output_value;
+            } else {
+                event.rounds[AVM_KECCAKF1600_NUM_ROUNDS - 1].state_chi[i][j] = output_value;
+            }
+        }
+    }
+
+    return event;
+}
+
+TEST(KeccakF1600TraceGenTest, MemorySliceReadAndWrite)
+{
+    TestTraceContainer trace;
+    KeccakF1600TraceBuilder builder;
+
+    // Create a standard event
+    auto event = create_standard_event_mem_slice();
+
+    // Process the event
+    builder.process_memory_slices({ event }, trace);
+
+    // Get the rows
+    auto rows = trace.as_rows();
+
+    // Check that we have the correct number of rows (2 * AVM_KECCAKF1600_STATE_SIZE for read and write)
+    EXPECT_EQ(rows.size(), 2 * AVM_KECCAKF1600_STATE_SIZE + 1); // +1 for the initial empty row
```

Comment:
nit: probably want this to be `ASSERT_EQ`, if this is false the future matchers are undefined since they're accessing specific indices

Reply:
same for other incidences of this